core/base/result.go

```diff
@@ -28,25 +29,38 @@ const (
 	BlockTypeCircuitBreaking
 	BlockTypeSystemFlow
 	BlockTypeHotSpotParamFlow
+	// BlockTypeRegistryStart convenient to expand the type to avoid conflicts
+	BlockTypeRegistryStart
 )
 
+var (
+	blockTypeNames = []string{
```

Comment:
Use slice does has the better performance, that's good!
But i have some concern about the maintainability, the order of const variable can not be changed, this is where potential bugs can arise.

Reply:
OK。In order to improve the scalability and facilitate the registration of new types, it is modified to map. Although the performance of map is poor, it is acceptable.

![image](https://user-images.githubusercontent.com/13758306/108508558-f867b700-72f6-11eb-9bc8-ce8374212c5d.png)

Reply:
Based on your bench, the performance of map is bad.
It's not suggested.

Reply:
Think of maintainability, map is acceptable.